src/main/java/com/blackducksoftware/integration/jira/task/JiraTaskTimed.java

```diff
@@ -263,4 +266,28 @@ private boolean checkUserInPluginGroups(final String jiraPluginGroupsString, fin
         }
         return false;
     }
+
+    private PluginConfigurationDetails updateOldMappingsIfNeeded(final PluginConfigurationDetails pluginConfigurationDetails, final PluginSettings pluginSettings) {
+        final BlackDuckJiraConfigSerializable config = new BlackDuckJiraConfigSerializable();
+        if (StringUtils.isNotBlank(pluginConfigurationDetails.getProjectMappingJson())) {
+            config.setHubProjectMappingsJson(pluginConfigurationDetails.getProjectMappingJson());
+            if (!config.getHubProjectMappings().isEmpty()) {
+                final BlackDuckProjectMapping blackDuckProjectMapping = config.getHubProjectMappings().iterator().next();
```

Comment:
I don't like using `.iterator().next()`; seems like a code smell. Can this be changed to: 
```
config.getHubProjectMappings()
                .stream()
                .findFirst();
                // And maybe .orElse(...) or .orElseThrow(...)
```